### PR TITLE
feat: reset preferences instead of deleting + add switch to use preferences in preferences + fix bug for unknown mandatory attributes

### PR DESCRIPTION
**Pull request by stephanegigandet**
- delete preferences button changed to reset preferences button + a precision that it will be Nutri-Score, Eco-Score, NOVA
- in the preferences selection, add a switch at the top to use the preferences for matching products
- if a mandatory attribute is unknown, the product status is set to unknown (bug identified by @monsieurtanuki in https://github.com/openfoodfacts/smooth-app/issues/1894#issuecomment-1132589366 )

image

### CODE REVIEW

File: po/common/en.po [hunk 1]
```diff
@@ -6155,3 +6155,13 @@ msgstr "May not match"
...
 msgid "Does not match"
 msgstr "Does not match"
+
+msgctxt "reset_preferences"
+msgid "Use default preferences"
+msgstr "Use default preferences"
+
+msgctxt "reset_preferences_details"
+msgid "Nutri-Score, Eco-Score and NOVA"
+msgstr "Nutri-Score, Eco-Score and NOVA"
```
Comment (teolemon) [inline po/common/en.po:6165]:
```suggestion
msgstr "Nutri-Score, Eco-Score and food processing level (NOVA)"
```

File: html/js/product-preferences.js [hunk 2]
```diff
@@ -84,6 +85,39 @@ function display_selected_preferences (target_selected_summary, product_preferen
...
+
+	var checked = '';
+	if (use_user_product_preferences_for_ranking) {
+		checked = " checked";
+	}	
+
+	var html = '<fieldset class="switch round success" tabindex="0" id="' + checkbox_id +'_switch" style="float:left;margin:0;margin-right:.5rem;padding-top:0.1rem;">'
+	+ '<input class="preferences_checkboxes" id="' + checkbox_id + '" type="checkbox"' + checked + '>'
+	+ '<label for="' + checkbox_id +'"></label></fieldset>'
+	+ '<label for="' + checkbox_id +'" style="float:left;margin-right:1em;padding-top:4px">' + preferences_text + '</label>';
```
Comment (alexgarel) [inline html/js/product-preferences.js:98]: One day you will embrace javascript template strings :-P

**Review by alexgarel — APPROVED** [review]
LGTM + tested ok

File: html/js/product-preferences.js (hunk 2 above)
Comment (stephanegigandet) [reply html/js/product-preferences.js:98]: I'll do it in another PR :)

**Pull request merged**